deployment/terraform/create.go

```diff
@@ -324,6 +332,175 @@ func (t *Terraform) Create(initData bool) error {
 	return nil
 }
 
+// checkCloudWatchLogsPolicy checks whether the AWS account has a
+// CloudWatchLogs resource-based policy attached to the OpenSearch service
+// that allows the logs:PutLogEventsBatch, logs:PutLogEvents and
+// logs:CreateLogStream actions over the OpenSearch service.
+// In particular, this is the minimum we need:
+//
+//	{
+//		Version: "2012-10-17",
+//		Statement: []{
+//			{
+//				Effect: "Allow",
+//				Principal: polDocStmtPrincipal{
+//					Service: "es.amazonaws.com",
+//				},
+//				Action: []string{
+//					"logs:PutLogEventsBatch",
+//					"logs:PutLogEvents",
+//					"logs:CreateLogStream",
+//				},
+//				Resource: "arn:aws:logs:*",
+//			},
+//		},
+//	}
+//
+// If such policy does not exist, we create it. Note that we are doing this
+// outside of Terraform. This is because we only need one policy in the
+// account, not one per deployment. This behaviour is also enforced by the rate
+// limitation on this type of policies: there can only be 10 such policies per
+// region per account. Check the docs for more information:
+// https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/cloudwatch_limits_cwl.html
+func (t *Terraform) checkCloudWatchLogsPolicy() error {
```

Comment:
This does more than just "check", so can we rename it to "createCloudWatchLogsPolicyIfNotExists"?